contributors/design-proposals/cluster-info-configmap.md

```diff
@@ -0,0 +1,25 @@
+# Persist cluster-wide data in ConfigMap
+
+## Abstract
+
+We want to allow cluster-wide data to be persisted for general purposes, currently for us (service catalog) is to focus on the ability to identify different clusters that access our service broker with a 'cluster-id'.
+
+## Motivation
+
+Currently there is no way for service catalog api server to identify the different clusters that access our service brokers. The goal of this design is to have a unify storage for cluster related information. The information should be retained and survive between component failures. 
+
+## Detailed design
+
+A configmap `cluster-info` is created in the `default` namespace at the starup of the controller. The initial content of the map will be as the following:
+
+```
+apiVersion: v1
+data:
+  cluster-id: uuid-of-the-cluster
+kind: ConfigMap
+metadata:
+  name: cluster-info
+  namespace: default
```

Comment:
- I don't see any reason why cluster-id needs to be private - there might be other cluster-wide data that should be private, but can deal with those in another PR when we have a need. So for now we'd like to keep this configMap simple and just let it be for global/public data.
- no opinion on which namespace it sits, but the default serviceAccount sits in 'default' (I think) so that's why we chose it